pkg/virt-dhcp/dnsmasq_test.go

```diff
@@ -0,0 +1,129 @@
+package virtdhcp
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var testdnsmasq *DNSmasqInstance
+var cmdArgs []string
+var cmdArgsPrefix string = "-k -d --strict-order --bind-dynamic --no-resolv --dhcp-no-override --dhcp-authoritative --except-interface=lo --dhcp-hostsfile=/tmp/dhcp.hosts --dhcp-leasefile=/tmp/dhcp.leases --pid-file=/tmp/dhcp.pid"
+
+type MockOSHandler struct {
+	OSHandler
+}
+
+func (o *MockOSHandler) writeDHCPHostsFile(file *os.File, dhcpHosts []string) error {
+	errRet := fmt.Errorf("failed to match dhcphosts")
+	if len(testdnsmasq.hosts) != len(dhcpHosts) {
+		return errRet
+	}
+	for _, host := range dhcpHosts {
+		data := strings.Split(host, ",")
+		if val, ok := testdnsmasq.hosts[data[1]]; ok {
+			lease, _ := strconv.Atoi(data[2])
+			if val.Mac != data[0] || val.Lease != lease {
+				return errRet
+			}
+		}
+	}
+
+	return nil
+}
+
+func (o *MockOSHandler) killProcessIfExist(pid int) error {
+	return nil
+}
+
+func (o *MockOSHandler) executeCommand(path string, startArgs []string) (*exec.Cmd, error) {
+	cmdArgs = startArgs
+	return &exec.Cmd{Process: &os.Process{Pid: 2509}}, nil
+}
+
+var _ = Describe("DHCP", func() {
+
+	BeforeEach(func() {
+		DHCPLeaseFile = "/tmp/dhcp.leases"
```

Comment:
We've been creating isolated temporary directories for test. That ensures we don't collide with anything already present on disk.
```
tmpDir, _ := ioutil.TempDir("", "cloudinittest")

AfterSuite(func() {
        os.RemoveAll(tmpDir)
})
```